### PR TITLE
Update deprecated usage of `tape` and `qtape` properties in `QNode`

### PR DESCRIPTION
**Pull request by andrijapau**
Summary:

Updating deprecated code that was introduced from a recent deprecation https://github.com/PennyLaneAI/pennylane/pull/6583.

[sc-76836]

### CODE REVIEW

File: demonstrations/qnspsa.py [hunk 1]
```diff
@@ -833,10 +828,8 @@ def __get_spsa_grad_tapes(self, cost, params):
         # used to estimate the gradient per optimization step. The sampled
         # direction is of the shape of the input parameter.
         direction = self.__get_perturbation_direction(params)
-        cost.construct([params + self.finite_diff_step * direction], {})
-        tape_forward = cost.tape.copy(copy_operations=True)
-        cost.construct([params - self.finite_diff_step * direction], {})
-        tape_backward = cost.tape.copy(copy_operations=True)
+        tape_forward = qml.workflow.construct_tape(cost)(*[params + self.finite_diff_step * direction])
+        tape_backward = qml.workflow.construct_tape(cost)(*[params - self.finite_diff_step * direction])
```
Comment (josh146) [inline demonstrations/qnspsa.py:832]: Hey @andrijapau, quick question; could this part of the tutorial be rewritten as a transform?

Comment (mudit2812) [reply demonstrations/qnspsa.py:832]: Imo it'll make the implementation longer and consequently, the demo less readable.

**Review by mudit2812 — CHANGES_REQUESTED** [review]
`qnspsa` demo's metadata.json file is missing an update.

**Review by mudit2812 — APPROVED** [review]
No blockers from my end so I'm approving, but you should probably wait to merge until the conversation Josh brought up is resolved.

File: demonstrations/qnspsa.py [hunk 2]
```diff
@@ -358,22 +358,17 @@ def get_grad(params_curr):
...
+    ops = tape_forward.operations + list(qml.adjoint(copy(op)) for op in reversed(tape_inv.operations))
 
     with qml.tape.QuantumTape() as tape:
-        for op in op_forward:
+        for op in ops:
             qml.apply(op)
-        for op in reversed(op_inv):
-            qml.adjoint(copy(op))
-        qml.probs(wires=qnode.tape.wires.labels)
+        qml.probs(wires=tape_forward.wires.labels)
```
Comment (albi3ro) [inline demonstrations/qnspsa.py:370]:
```
tape = qml.tape.QuantumTape(ops, [qml.probs(wires=tape_forwrad.wires)])
```

File: demonstrations/qnspsa.py (hunk 1 above)
Comment (albi3ro) [reply demonstrations/qnspsa.py:832]: I would say out of scope for a quick deprecation change.

File: demonstrations/qnspsa.py [hunk 3]
```diff
@@ -358,24 +358,12 @@ def get_grad(params_curr):
...
-        for op in op_forward:
-            qml.apply(op)
-        for op in reversed(op_inv):
-            qml.adjoint(copy(op))
-        qml.probs(wires=qnode.tape.wires.labels)
-    return tape
+    tape_forward = qml.workflow.construct_tape(qnode)(*params1)
+    tape_inv = qml.workflow.construct_tape(qnode)(*params2)
 
+    ops = tape_forward.operations + list(qml.adjoint(copy(op)) for op in reversed(tape_inv.operations))
```
Comment (albi3ro) [inline demonstrations/qnspsa.py:365]:
```suggestion
    ops = tape_forward.operations + list(qml.adjoint(op) for op in reversed(tape_inv.operations))
```
Sorry... just realized we don't need the copy.

Comment (andrijapau) [reply demonstrations/qnspsa.py]: Done.

**Review by albi3ro — APPROVED** [review]
👍

**Pull request merged**